components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -342,5 +338,30 @@ def main(argv):
     return 0
 
 
+def _get_ipv4_address() -> str | None:
+    """
+    Retrieves an IPv4 address of the host for network communication.
+
+    :returns: The first non-loopback IPv4 address it finds.
+    If no non-loopback address is available, returns the first loopback IPv4 address.
+    If no IPv4 address is found, returns None.
+    """
+    fallback_ip = None
+
+    for _, addresses in psutil.net_if_addrs().items():
```

Comment:
Sorry. Should have caught this in the previous review.

```suggestion
    for addresses in psutil.net_if_addrs().values():
```